### ISSUE TITLE
ssh will meet error when upgrade if too much component deployed in one node

### ISSUE THREAD

**Issue by lance6716** [pingcap/tiup#1478]
## Bug Report

Please answer these questions before submitting your issue. Thanks!

1. What did you do?

DM has a limit that number of DM-worker must be greater than number of upstream sources, so user may deploy lots of DM-worker in one node. And user may need upgrade the version of DM.

2. What did you expect to see?

upgrade successfully.

3. What did you see instead?

meet ssh error like

```
--ssh system
Error: test -d /home/tidb/dm/deploy/dm-worker-28067/bin.old.v2.0.0 || cp -r /home/tidb/dm/deploy/dm-worker-28067/bin /home/tidb/dm/deploy/dm-worker-28067/bin.old.v2.0.0: executor.ssh.execute_failed: Failed to execute command over SSH for 'tidb@172.16.5.156:22' {ssh_stderr: ssh_exchange_identification: Connection closed by remote host
, ssh_stdout: , ssh_command: export LANG=C; PATH=$PATH:/usr/bin:/usr/sbin test -d /home/tidb/dm/deploy/dm-worker-28067/bin.old.v2.0.0 || cp -r /home/tidb/dm/deploy/dm-worker-28067/bin /home/tidb/dm/deploy/dm-worker-28067/bin.old.v2.0.0}, cause: exit status 255
--ssh builtin
Error: test -d /home/tidb/dm/deploy/dm-worker-28098/bin.old.v2.0.0 || cp -r /home/tidb/dm/deploy/dm-worker-28098/bin /home/tidb/dm/deploy/dm-worker-28098/bin.old.v2.0.0: executor.ssh.execute_failed: Failed to execute command over SSH for
'tidb@172.16.6.57:22' {ssh_stderr: , ssh_stdout: , ssh_command: export LANG=C; PATH=$PATH:/usr/bin:/usr/sbin test -d /home/tidb/dm/deploy/dm-worker-28098/bin.old.v2.0.0 || cp -r /home/tidb/dm/deploy/dm-worker-28098/bin /home/tidb/dm/deplo
y/dm-worker-28098/bin.old.v2.0.0}, cause: ssh: handshake failed: EOF
```

Now we can use `tiup dm audit` and `tiup dm replay` to retry, but not a good experience.

4. What version of TiUP are you using (`tiup --version`)?

**Comment by AstroProfundis** [comment]
That's a limit of SSH itself, `replay` is designed to solve this problem.

Also, #1420 may be helpful after released in `v1.6.0`

**Comment by lance6716** [comment]
> That's a limit of SSH itself, `replay` is designed to solve this problem.
> 
> Also, #1420 may be helpful after released in `v1.6.0`

thanks, will test later

**Comment by lance6716** [comment]
confirmed by our user
